### PR TITLE
update lib api types

### PR DESCRIPTION
**Pull request by Eein**
## What
pnpm generate:api

## Link to Issue

n/a

### Steps for testing

None required

### Images

```js
import type ...
```

### CODE REVIEW

**Review by TheTedder — APPROVED** [review]

File: lib/api/data-contracts.ts [hunk 1]
```diff
@@ -266,6 +272,12 @@ export type TimedRunViewModel = BaseRunViewModel & {
   time: string
 }
 
+export interface UpdateLeaderboardRequest {
+  name?: string
+  slug?: string
+  info?: string
+}
```
Comment (zysim) [inline lib/api/data-contracts.ts:278]: So this request rejects if we've got all fields empty. Would we want to do this instead?

```ts
export type UpdateLeaderboardRequest = {
  name: string,
  slug?: string,
  info?: string
} | {
  name?: string,
  slug: string,
  info?: string
} | {
  name?: string,
  slug?: string,
  info: string
}
```

Probably too wordy though, right

Comment (erunks) [reply lib/api/data-contracts.ts:278]: Ideally this is something we'd just have to handle from wherever we make this request. Currently the `lib/api` folder and its contents are completely auto-generated by an external package. So anything we'd change here would have to be kept manually up to date on each update
